src/main/java/com/example/demo/repository/SpatialRelationshipsRepository.java

```diff
@@ -0,0 +1,11 @@
+package com.example.demo.repository;
+
+import java.util.Optional;
+
+public interface SpatialRelationshipsRepository {
+    Optional<Boolean> isIntersects(long lineId, long polygonId);
```

Comment:
I guess we have to inform the user which kind of param is not correct.
Not need Optional. Use throw exception and provide message for user